src/main/java/org/apache/mesos/scheduler/plan/DefaultPhase.java

```diff
@@ -15,161 +19,82 @@
-        public List<Block> getBlocks() {
-            return blocks;
-        }
-
-        public Builder addBlock(final Block block) {
-            if (block != null) {
-                this.blocks.add(block);
-            }
-            return this;
-        }
-
-        public Builder removeBlock(final Block block) {
-            if (block != null) {
-                this.blocks.remove(block);
-            }
-            return this;
-        }
-
-        public Builder addBlocks(Collection<Block> blocks) {
-            if (blocks != null) {
-                this.blocks.addAll(blocks);
-            }
-            return this;
-        }
-
-        public Builder clearBlocks() {
-            this.blocks.clear();
-            return this;
-        }
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final UUID id = UUID.randomUUID();
+    private final String name;
+    private final Strategy<Block> strategy;
+    private final List<String> errors;
+    private final List<Block> blocks;
 
-        public UUID getId() {
-            return id;
-        }
+    public DefaultPhase(String name, List<Block> blocks, Strategy<Block> strategy, List<String> errors) {
+        this.name = name;
+        this.blocks = blocks;
+        this.strategy = strategy;
+        this.errors = errors;
+    }
 
-        public Builder setId(UUID id) {
-            if (id != null) {
-                this.id = id;
-            }
-            return this;
-        }
+    @Override
+    public Strategy<Block> getStrategy() {
+        return strategy;
+    }
 
-        public String getName() {
-            return name;
-        }
+    @Override
+    public UUID getId() {
+        return id;
+    }
 
-        public Builder setName(String name) {
-            if (name != null) {
-                this.name = name;
-            }
-            return this;
-        }
+    @Override
+    public String getName() {
+        return name;
+    }
 
-        public DefaultPhase build() {
-            return DefaultPhase.create(id, name, blocks);
+    @Override
+    public Status getStatus() {
+        if (getStrategy().isInterrupted()) {
+            return Status.WAITING;
         }
-    }
 
-    /**
-     * Static factory method.
-     * @param id The unique identifier for the Phase.
-     * @param name The name of the Phase.
-     * @param blocks The blocks contained in the Phase.
-     * @return A new DefaultPhase constructed from id and name and containing
-     * blocks.
-     */
-    public static DefaultPhase create(
-            UUID id, String name, Collection<? extends Block> blocks) {
-        return new DefaultPhase(id, name, blocks);
+        return PlanUtils.getStatus(getChildren());
     }
 
-    public static Builder builder() {
-        return new Builder();
+    @Override
+    public void setStatus(Status status) {
+        logger.warn("Setting status does not effect status. Status is derived from Strategy and child statuses.");
```

Comment:
We can do that.  We'll just have to be careful not to call this, especially through HTTP endpoints.  Or I'm thinking we could set the same status on all child elements...  more consistent with the `forceComplete` and `restart` behavior.